app/src/main/java/org/wikipedia/analytics/InstallReferrerListener.java

```diff
@@ -0,0 +1,175 @@
+package org.wikipedia.analytics;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.text.TextUtils;
+
+import androidx.annotation.NonNull;
+
+import com.android.installreferrer.api.InstallReferrerClient;
+import com.android.installreferrer.api.InstallReferrerStateListener;
+
+import org.wikipedia.WikipediaApp;
+import org.wikipedia.page.PageActivity;
+import org.wikipedia.settings.Prefs;
+import org.wikipedia.util.ShareUtil;
+import org.wikipedia.util.log.L;
+
+/**
+ * To test the Google Play Install Referrer functionality:
+ *
+ * - Make sure the app is uninstalled.
+ * - Launch the Play Store via the usual specially crafted campaign link, such as:
+ *   https://play.google.com/store/apps/details?id=org.wikipedia&referrer=utm_source%3Dtest_source%26utm_medium%3Dtest_medium%26utm_term%3Dtest-term%26utm_content%3Dtest_content%26utm_campaign%3Dtest_name
+ *
+ * - ...But do NOT click the "install" button in the Play Store to install the app.
+ * - Launch or debug the app in the usual way.
+ * - The Install Referrer service should work, and will pass through the correct referrer.
```

Comment:
I've only seen once when launching the dev app. Does it happen every time after clicked the link and then launch the app?

Reply:
Ideally you should test this with the Production flavor, since that's what the Play Store knows about.

Reply:
I see! You're right!